src/main/java/com/jcabi/aether/RepositoryAuthentication.java

```diff
@@ -67,11 +67,16 @@
      * Creates a new authentication with the specified properties.
      * @param auth The authentication object.
      */
+    @SuppressWarnings("PMD.ConfusingTernary")
     public RepositoryAuthentication(final Authentication auth) {
         this.username = auth.getUsername();
-        this.password = auth.getPassword().toCharArray();
+        // @checkstyle AvoidInlineConditionalsCheck (2 lines)
+        this.password = auth.getPassword() != null
```

Comment:
@adko-pl does the addition of a proxy make this null check necessary? or is this change unrelated to the issue?